src/storm/storage/expressions/SimplificationVisitor.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#include <stack>
+
+#include "storm/storage/expressions/Expression.h"
+#include "storm/storage/expressions/ExpressionVisitor.h"
+
+namespace storm {
+    namespace expressions {
+        class SimplificationVisitor : public ExpressionVisitor {
```

Comment:
@tquatmann  Arguably, simplify should be merged into this visitor; the fact that we can use state here means that we can do much more static analysis.  I will rename for now.

Reply:
Yeah, that would also be possible. But then this also needs a way to configure what kind of operators we want to replace (maybe I'm fine with certain operators and want to keep them to avoid blowing up the expression)

Reply:
But that is a general concern that I also may have with simplify right now? As in, it is not clear what a simplification is and what not?

Reply:
Fair enough, It's totally unclear what kind of simplifications are allowed with `simplify`.
I'd say `simplify` tries to reduce the size of the expression whereas the `SimplificationVisitor` actually might increase the size of the expression but uses a simpler set of operators.
These are two different kinds of simplification and the operator-simplification might conflict with the size-simplification. I wouldn't want to increase the size of my expression just to eliminate an operator that my implementation supports anyway (e.g. it's relatively easy to add support for new expression operators in the sparse engine but it can be quite a pain for DD)

Reply:
Fully agree. I think that the rename should avoid the confusion for the time being, and I am in the course of adding simplifications to predicate expressions (there are quite a few options).